### PR TITLE
[Issue#59] install ESLint lib for static code analysis

### LINKED ISSUE #59: [Project Config] install ESLint lib for static code analysis
## ✏️ Description
Install [ESLint](https://eslint.org/) library to help with enforcing consistent coding practices

## 🔧 Why We Need This
Alongside the already installed [Prettier](https://prettier.io/) lib, it will be a great tool for static code analysis that finds any code issues

## 🚀 Proposed Solution
- install `ESlint` lib as per description from jaktestowac course
- install prettier plugin for sorting imports: `prettier-plugin-sort-imports` as per description from jaktestowac course
lessons 03 && 04: 
https://jaktestowac.pl/lesson/pw2s01l03/
https://jaktestowac.pl/lesson/pw2s01l04/ 

## ✅ Issue reporting checklist

- [x] I created a descriptive title for this issue
- [x] I checked that it hasn't been reported yet
- [x] I assigned the issue or added `help-wanted` label
- [x] I added appropriate labels

### PR DESCRIPTION
**Pull request by KarolinaZakrzewska**
### Scope of changes:
- installed Eslint plugins: `eslint-plugin`, `eslint-parser`, `eslint-config-prettier`, `eslint-plugin-playwright`, `eslint-plugin-prettier`,
- updated ignore files: `.eslintignore`, `.prettieringore`, `.gitignore`,
- created eslint configuration file: `.eslintrc.json`,
- updated prettier configuration file: `.prettierrs.json`,
- after running prettier and linter, all files that require re-formatting were adjusted,
- removed unnecessary comments and disabled eslint no-var-requires/no-used-vars errors in playwright config file: `playwright.config.ts`,
- added scripts for running Eslint and Prettier in package.json

### Additional, not related changes:
- added 2 new folders for E2E tests: components and pages using .gitkeep file,
- replaced expectedStatusCode var with a proper enumeration constant,
- added scripts for running tests in UI mode

### How to use Prettier?
* to run prettier use the following command: 
```
npm run format
```

* to run lint use the following command:
```
npm run lint
```
---
I have ensured that::
- [X] The PR is associated with the relevant Issue before its creation.
- [x] All new and existing tests passed
- [x] My commit messages follow the `kawqa-gad-playwright` project's [git commit message format]https://github.com/kat-kan/kawqa-gad-playwright/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bugITwhisperer** [comment]
I've just completed the outstanding PR that I took over from @KarolinaZakrzewska :) 
@kat-kan , @Justyna-KO : could you take a look please?

File: .github/workflows/pw-tests.yml [hunk 1]
```diff
@@ -2,39 +2,39 @@ name: Run Playwright API Auto-Checks on CI
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: npm ci
+      - name: Install Playwright Chromium Browser
+        run: npx playwright install --with-deps chromium
+      - name: Create .env file
+        run: |
+          echo "BASE_URL='http://localhost:3000'" > .env
```
Comment (kat-kan) [inline .github/workflows/pw-tests.yml:25]: @bugITwhisperer would you like to try using secrets (as we initially discussed during the last meeting)? 
I just created ones for our repository and named them exactly like here (BASE_URL, USER_EMAIL, USER_PASSWORD)
Docs: https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions

**Review by kat-kan — APPROVED** [review]
To keep our docs nice and clean, maybe it's worth a shot to update unfished PR description?>

File: .github/workflows/pw-tests.yml (hunk 1 above)
Comment (bugITwhisperer) [reply .github/workflows/pw-tests.yml:25]: @kat-kan : yeah, definitely - just like we discussed during our last meeting! 😄 However, I will address this issue in a separate task though - this one: https://github.com/kat-kan/kawqa-gad-playwright/issues/75

**Review by kat-kan — APPROVED** [review]

**Review by Justyna-KO — APPROVED** [review]

File: .github/workflows/pw-tests.yml [hunk 2]
```diff
@@ -2,39 +2,39 @@ name: Run Playwright API Auto-Checks on CI
 
 on:
   push:
-    branches: [ "main" ]
+    branches: ['main']
```
[the remark this answers is not in the document]
Comment (bugITwhisperer) [reply .github/workflows/pw-tests.yml]: It's a great idea! Added here: https://github.com/kat-kan/kawqa-gad-playwright/pull/66/commits/c17417e9ff9abf798b10cb8d8be54ed657bf0971

**Review by Justyna-KO — APPROVED** [review]

**Pull request merged**